src/publish/mod.rs

```diff
@@ -31,25 +31,48 @@ impl CaPublisher {
     }
 
     pub async fn publish(&self, ca_handle: &Handle, actor: &Actor) -> Result<(), Error> {
-        let ca = self.caserver.get_ca(ca_handle).await?;
+        // Since this is called by the scheduler, this acts as a no-op for new CAs which do not yet have any repository configured.
+        for (contact, elements) in self.caserver.ca_repo_elements(ca_handle).await? {
+            self.sync_repo(ca_handle, contact, elements, actor).await?;
+        }
 
-        // Since this is called by the scheduler, this should act as a no-op for
-        // new CAs which do not yet have any repository configured.
-        let repo_contact = match ca.repository_contact() {
-            Ok(repo) => repo,
-            Err(_) => return Ok(()),
-        };
+        // Best effort clean-up of old repos
+        for deprecated in self.caserver.ca_take_deprecated_repos(ca_handle)? {
+            info!(
+                "Will try to clean up deprecated repository '{}' for CA '{}'",
+                deprecated, ca_handle
+            );
+            if self.sync_repo(ca_handle, deprecated, vec![], actor).await.is_err() {
+                info!(
+                    "Could not clean up deprecated repository. This is fine - objects there are no longer referenced."
```

Comment:
This message sounds like there was a failure of some kind and with failures one wonders why did it fail, is something else wrong as well? Is it that in this case it doesn't matter because the failure is not with this Krill instance but is due to the external repo not being available or something? I find this log message a bit scary, maybe it can be improved.